Cabal/Distribution/Compat/Prelude.hs

```diff
@@ -117,7 +118,12 @@ import Control.DeepSeq               (NFData (..))
 import Data.Data                     (Data)
 import Data.Typeable                 (Typeable)
 import Distribution.Compat.Binary    (Binary (..))
-import Distribution.Compat.Semigroup (Semigroup (..), gmappend, gmempty)
+#if !MINVER_base_411 && MINVER_base_48
```

Comment:
don't we need to import `Semigroup` always before base-4.11, let's see what travis says

Reply:
I've made quite a lot of these changes today, so my head is spinning. I think you're right, I'm having a closer look.

Reply:
On closer investigation, `import Distribution.Compat.Semigroup (Semigroup (..))` **is** used because it is rexported as `Semigroup (..)`. Note that `Semigroup (..)` from `Distribution.Compat.Semigroup` contains only `Semigroup((<>))`. The import is still redundant (in newer base version) because the default Prelude is imported (non-qualified) which exports `Semigroup((<>))`.

I'll try and resolve the warning by hiding `Semigroup((<>))`  from prelude.